cases.json

```diff
@@ -1314,101 +1486,575 @@
+        "v": "1.2.3+build.abc.123"
+      },
+      true
+    ],
+    [
+      "$vne - pass",
+      {
+        "v": {
+          "$vne": "1.2.3"
+        }
+      },
+      {
+        "v": "2.2.3"
+      },
+      true
+    ],
+    [
+      "$vne - pass (prerelease)",
+      {
+        "v": {
+          "$vne": "1.2.3"
+        }
+      },
+      {
+        "v": "1.2.3-alpha"
+      },
+      true
+    ]
+  ],
+  "versionCompare": {
+    "lt": [
+      ["0.9.99", "1.0.0", true],
+      ["0.9.0", "0.10.0", true],
+      ["1.0.0-0.0", "1.0.0-0.0.0", true],
+      ["1.0.0-9999", "1.0.0--", true],
+      ["1.0.0-99", "1.0.0-100", true],
+      ["1.0.0-alpha", "1.0.0-alpha.1", true],
+      ["1.0.0-alpha.1", "1.0.0-alpha.beta", true],
+      ["1.0.0-alpha.beta", "1.0.0-beta", true],
+      ["1.0.0-beta", "1.0.0-beta.2", true],
+      ["1.0.0-beta.2", "1.0.0-beta.11", true],
+      ["1.0.0-beta.11", "1.0.0-rc.1", true],
+      ["1.0.0-rc.1", "1.0.0", true],
+      ["1.0.0-0", "1.0.0--1", true],
+      ["1.0.0-0", "1.0.0-1", true],
+      ["1.0.0-1.0", "1.0.0-1.-1", true]
+    ],
+    "gt": [
+      ["0.0.0", "0.0.0-foo", true],
+      ["0.0.1", "0.0.0", true],
+      ["1.0.0", "0.9.9", true],
+      ["0.10.0", "0.9.0", true],
+      ["0.99.0", "0.10.0", true],
+      ["2.0.0", "1.2.3", true],
+      ["v0.0.0", "0.0.0-foo", true],
+      ["v0.0.1", "0.0.0", true],
+      ["v1.0.0", "0.9.9", true],
+      ["v0.10.0", "0.9.0", true],
+      ["v0.99.0", "0.10.0", true],
+      ["v2.0.0", "1.2.3", true],
+      ["0.0.0", "v0.0.0-foo", true],
+      ["0.0.1", "v0.0.0", true],
+      ["1.0.0", "v0.9.9", true],
+      ["0.10.0", "v0.9.0", true],
+      ["0.99.0", "v0.10.0", true],
+      ["2.0.0", "v1.2.3", true],
+      ["1.2.3", "1.2.3-asdf", true],
+      ["1.2.3", "1.2.3-4", true],
+      ["1.2.3", "1.2.3-4-foo", true],
+      ["1.2.3-5-foo", "1.2.3-5", true],
+      ["1.2.3-5", "1.2.3-4", true],
+      ["1.2.3-5-foo", "1.2.3-5-Foo", true],
+      ["3.0.0", "2.7.2+asdf", true],
+      ["1.2.3-a.10", "1.2.3-a.5", true],
+      ["1.2.3-a.b", "1.2.3-a.5", true],
+      ["1.2.3-a.b", "1.2.3-a", true],
+      ["1.2.3-a.b.c", "1.2.3-a.b.c.d", false],
+      ["1.2.3-a.b.c.10.d.5", "1.2.3-a.b.c.5.d.100", true],
+      ["1.2.3-r2", "1.2.3-r100", true],
+      ["1.2.3-r100", "1.2.3-R2", true],
+      ["a.b.c.d.e.f", "1.2.3", true],
+      ["10.0.0", "9.0.0", true],
+      ["10000.0.0", "9999.0.0", true]
+    ],
+    "eq": [
+      ["1.2.3", "1.2.3", true],
+      ["1.2.3", "v1.2.3", true],
+      ["1.2.3-0", "v1.2.3-0", true],
+      ["1.2.3-1", "1.2.3-1", true],
+      ["1.2.3-1", "v1.2.3-1", true],
+      ["1.2.3-beta", "1.2.3-beta", true],
+      ["1.2.3-beta", "v1.2.3-beta", true],
+      ["1.2.3-beta+build", "1.2.3-beta+otherbuild", true],
+      ["1.2.3-beta+build", "v1.2.3-beta+otherbuild", true],
+      ["1-2-3", "1.2.3", true],
+      ["1-2-3", "1-2.3+build99", true],
+      ["1-2-3", "v1.2.3", true],
+      ["1.2.3.4", "1.2.3-4", true]
+    ]
+  },
```

Comment:
Are there tests for this? There are some conditions in this that aren't covered in the other ones.

Reply:
It looks like json_test.go was collapsed in the diff. I see there's tests for it. Marking as resolved.